src/Imath/ImathMatrix.h

```diff
@@ -3574,22 +3574,22 @@ template <class S>
 IMATH_HOSTDEVICE IMATH_CONSTEXPR14 inline Matrix44<T>&
                  Matrix44<T>::setValue (const Matrix44<S>& v) IMATH_NOEXCEPT
 {
-    x[0][0] = v.x[0][0];
-    x[0][1] = v.x[0][1];
-    x[0][2] = v.x[0][2];
-    x[0][3] = v.x[0][3];
-    x[1][0] = v.x[1][0];
-    x[1][1] = v.x[1][1];
-    x[1][2] = v.x[1][2];
-    x[1][3] = v.x[1][3];
-    x[2][0] = v.x[2][0];
-    x[2][1] = v.x[2][1];
-    x[2][2] = v.x[2][2];
-    x[2][3] = v.x[2][3];
-    x[3][0] = v.x[3][0];
-    x[3][1] = v.x[3][1];
-    x[3][2] = v.x[3][2];
-    x[3][3] = v.x[3][3];
+    x[0][0] = static_cast<T>(v.x[0][0]);
+    x[0][1] = static_cast<T>(v.x[0][1]);
+    x[0][2] = static_cast<T>(v.x[0][2]);
+    x[0][3] = static_cast<T>(v.x[0][3]);
+    x[1][0] = static_cast<T>(v.x[1][0]);
+    x[1][1] = static_cast<T>(v.x[1][1]);
+    x[1][2] = static_cast<T>(v.x[1][2]);
+    x[1][3] = static_cast<T>(v.x[1][3]);
+    x[2][0] = static_cast<T>(v.x[2][0]);
+    x[2][1] = static_cast<T>(v.x[2][1]);
+    x[2][2] = static_cast<T>(v.x[2][2]);
+    x[2][3] = static_cast<T>(v.x[2][3]);
+    x[3][0] = static_cast<T>(v.x[3][0]);
+    x[3][1] = static_cast<T>(v.x[3][1]);
+    x[3][2] = static_cast<T>(v.x[3][2]);
+    x[3][3] = static_cast<T>(v.x[3][3]);
```

Comment:
I think this block here is safe in for float(v.x[0][0]) context.

Reply:
Updated.

Reply:
Did you mean to add a commit?

Reply:
Ahem yes  ... *really* updated now :)